src/structurizr/workspace.py

```diff
@@ -195,18 +195,54 @@ def __init__(
 
     @classmethod
     def load(cls, filename: Union[str, Path]) -> "Workspace":
-        """"""
+        """Load a workspace from a file, which may be zipped."""
         filename = Path(filename)
         try:
             with gzip.open(filename, "rt") as handle:
-                ws_io = WorkspaceIO.parse_raw(handle.read())
+                return cls.loads(handle.read())
         except FileNotFoundError as error:
             raise error
         except OSError:
             with filename.open() as handle:
-                ws_io = WorkspaceIO.parse_raw(handle.read())
+                return cls.loads(handle.read())
+
+    @classmethod
+    def loads(cls, json_string: str) -> "Workspace":
+        """Load a workspace from a JSON string."""
+        ws_io = WorkspaceIO.parse_raw(json_string)
         return cls.hydrate(ws_io)
 
+    def dump(
+        self,
+        filename: Union[str, Path],
+        *,
+        zip: bool = False,
```

Comment:
A common pattern is to basically check `filename.endswith(".gz")` and then automatically gzip. This is a bit more versatile but less convenient. What do you think?

Reply:
I like the convenience of defaulting based on the filename, but still feel like we should give the caller control (especially being able to zip without `.gz` ending if they want.  I've refactored to support both - see what you think.